src/AppBundle/Controller/Legislatives/MapController.php

```diff
@@ -21,8 +22,21 @@ public function mapAction(): Response
     {
         $this->disableInProduction();
 
-        return $this->render('legislatives/map.html.twig', [
-            'candidates' => $this->getDoctrine()->getRepository(LegislativeCandidate::class)->findAll(), // @todo
-        ]);
+        return $this->render('legislatives/map.html.twig');
+    }
+
+    /**
+     * @Route("/api/candidats", name="api_legislatives_candidates")
+     * @Method("GET")
+     */
+    public function getCandidatesList(Request $request): JsonResponse
+    {
+        $this->disableInProduction();
+
+        if (!$request->isXmlHttpRequest()) {
+            throw $this->createNotFoundException();
```

Comment:
You can use `condition="request.isXmlHttpRequest"` in the route definition instead.